### PR TITLE
Make ProjectsController#destroy use background worker + service

### PR DESCRIPTION
**Pull request by oliverguenther**
Trying to delete large projects in OpenProject may result in timeout errors since currently, the deletion is performed in the context of the web request.

This PR introduces a `Projects::DeleteService` and contract and a delayed job to schedule the deletion. The services handles both delayed and non-delayed cases to ensure authorization can happen in one place.

It also refactors `UserMailer` to allow a separate `ProjectMailer` to be created to the size of UserMailer. The `BaseMailer` now inherits some OpenProject specifics regarding mail ID generation and whatnot.

https://community.openproject.com/wp/29214

### CODE REVIEW

**Comment by oliverguenther** [comment]
legacy_spec failing are the unrelated often-failing `git_repository_spec`

File: app/services/projects/delete_project_service.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    def initialize(user:, project:)
+      self.user = user
+      self.project = project
+      self.contract_class = ::Projects::DeleteContract
+    end
+
+    ##
+    # Deletes the given user if allowed.
+    #
+    # @return True if the user deletion has been initiated, false otherwise.
```
Comment (ulferts) [inline app/services/projects/delete_project_service.rb]: `s/user/project`

**Review by ulferts — CHANGES_REQUESTED** [review]
Code looks good and I aside from a type, I liked all of it.

Is there any chance, we can get an integration spec for the functionality?

**Comment by oliverguenther** [comment]
Added, thanks!

**Pull request merged**